Kokkos configuration failure with generate_makefile script

I am unable to configure Kokkos at SHA 1fe6f1b025fc2fa52c80ab70283b52897cbc145b with Cuda versions 9.2 nor 10.1 with the configure line below. I confirmed the configure line works with 3.2.00 release as a sanity check.

## CMake line

```bash
module load sems-env sems-cmake/3.12.2 sems-cuda/9.2 sems-gcc/5.3.0 sems-git

cmake -DCMAKE_CXX_COMPILER=$HOME/kokkos/bin/nvcc_wrapper -DCMAKE_CXX_FLAGS="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized" -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_ARCH_SNB=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DKokkos_CXX_STANDARD=14 -DCMAKE_BUILD_TYPE=RELEASE $HOME/kokkos
```

Full configuration output:

```bash
-- The CXX compiler identification is GNU 5.3.0
-- Check for working CXX compiler: /ascldap/users/ndellin/kokkos/bin/nvcc_wrapper
-- Check for working CXX compiler: /ascldap/users/ndellin/kokkos/bin/nvcc_wrapper -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - failed
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Kokkos
-- Compiler Version: 10.1.243
-- Using -std=c++14 for C++14 standard as feature
CMake Error at cmake/kokkos_test_cxx_std.cmake:53 (MESSAGE):
  Compiler NVIDIA should support cxx_std_14, but CMake reports feature not
  supported
Call Stack (most recent call first):
  cmake/kokkos_test_cxx_std.cmake:69 (kokkos_set_cxx_standard_feature)
  cmake/kokkos_tribits.cmake:205 (INCLUDE)
  CMakeLists.txt:168 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


-- Configuring incomplete, errors occurred!
See also "/ascldap/users/ndellin/kokkos/testing/Cuda9_OMP/CMakeFiles/CMakeOutput.log".
See also "/ascldap/users/ndellin/kokkos/testing/Cuda9_OMP/CMakeFiles/CMakeError.log".
```

--- a/bin/nvcc_wrapper
+++ b/bin/nvcc_wrapper
@@ -242,7 +242,7 @@ do
     fi
     shared_args="$shared_args $std_flag"
     ;;
-  --std=c++14|-std=c++14)
+  --std=c++11|-std=c++11|--std=c++14|-std=c++14)
     if [ -n "$std_flag" ]; then
        warn_std_flag
        shared_args=${shared_args/ $std_flag/}
@@ -252,7 +252,7 @@ do
     ;;
 
   #convert PGI standard flags to something nvcc can handle
-  --c++14|--c++17)
+  --c++11|--c++14|--c++17)
     if [ -n "$std_flag" ]; then
        warn_std_flag
        shared_args=${shared_args/ $std_flag/}

